src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -146,6 +147,77 @@ public void end(boolean interrupted) {
         };
     }
 
+    public CommandBase setAbsoluteAngle(final Rotation2d angle, final double speed) {
+
+        Drive thisDrive = this;
+
+        return new CommandBase() {
+            {
+                addRequirements(thisDrive);
+                setName("Absolute angle");
+            }
+            private double speed2 = speed;
+
+            @Override
+            public void initialize() {
+                speed2 *= Math.signum(angle.getDegrees() - pose.getRotation().times(1).getDegrees());
+            }
+
+            @Override
+            public void execute() {
+                updateSwerveSpeedAngle(() -> 0., () -> 0., () -> speed2);
+            }
+
+            @Override
+            public boolean isFinished() {
+                return Math.abs(pose.getRotation().times(1).getDegrees() - angle.getDegrees()) <= ROTATION_BUFFER;
+            }
+
+            @Override
+            public void end(boolean interrupted) {
+                updateSwerveSpeedAngle(() -> 0, () -> 0, () -> 0);
+            }
+
+        };
+    }
+
+    public CommandBase setRelitiveAngle(final Rotation2d angle, final double speed) {
+
+        Drive thisDrive = this;
+
+        return new CommandBase() {
+            {
+                addRequirements(thisDrive);
+                setName("Relitive Angle");
+            }
+            private double speed2 = speed;
+            private Rotation2d targetRotation;
```

Comment:
Since this entire variable is set once, why not create it with `thisDrive` up above?